### DISCUSSION TITLE
Git and GitHub setup

### DISCUSSION THREAD

**Discussion by brunj7** [microsoft/PlanetaryComputer#92] [Q&A]
Hi,

What is the recommended setup on the Planetary Hub regarding version control? ssh key, GitHub token? and if the latter how to best cache credentials?

Thanks for any tips!
Julien

**Comment by TomAugspurger** [comment]
There is [jupyterlab-git](https://github.com/jupyterlab/jupyterlab-git), but I'm not 100% sure how it manages credentials on JupyterHub deployments. I don't know if jupyterlab-git stores credentials on the filesystem / in-memory / in the browser.

The potential risk, as outlined in https://github.com/jupyterlab/jupyterlab-git/issues/1044#issuecomment-933157115, that a JupyterHub admin on the deployment (currently just me) can access files from your home directory, so it isn't best-practice to store secrets in the home directory.

If you [deploy your own hub](https://planetarycomputer.microsoft.com/docs/concepts/hub-deployment/) then you have more control over that sort of thing, and so jupyterlab-git might be a good option. If jupyterlab-git only stores the credentials in-memory then perhaps that's an OK compromise between security and convenience, at least enough to include jupyterlab-git in the image so that users can decide for themselves.
